packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -119,7 +119,10 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
      * @param error - optional error object to log
      */
     public sendTelemetryEvent(event: ITelemetryGenericEvent, error?: any) {
-        const newEvent: ITelemetryBaseEvent = { ...event, category: event.category ? event.category : "generic" };
+        const newEvent: ITelemetryBaseEvent = {
+            ...event,
+            category: event.category ?? error === undefined ?  "generic" : "error",
```

Comment:
this seems like better behavior to me, still allow the caller to override, but set category error if error is provided

Reply:
this makes it easy to log correctly when error is maybe undefined for the caller, and they don't need to muck with category, and in generall i'd rather over categorize things as error, as if we under categorize we will miss things

Reply:
Works for me. I mean, it's annoying that these go to different tables, but we already deal with that in how we query Kusto so whatever.